src/services/BaseAPIServices.ts

```diff
@@ -0,0 +1,467 @@
+import axios, { AxiosError, ResponseType } from 'axios';
+import { authenticateSilently } from './auth';
+import { metricKeys, metricStatus, track } from './appInsights';
+import * as FileSystem from 'expo-file-system';
+import * as Device from 'expo-device';
+
+export type BaseResource = {
+  scopes: string[];
+  apiBaseUrl: string;
+  subscriptionKey?: string;
+};
+
+export type BaseAPIOptions = {
+  authenticate?: boolean;
+  headers?: Record<string, any>;
+  responseType?: ResponseType;
+};
+
+export type DownloadFileOptions = BaseAPIOptions & {
+  filePath?: string;
+};
+
+const defaultOptions = {
+  authenticate: true,
+};
+
+const defaultFileDownloadOptions = {
+  ...defaultOptions,
+  filePath: FileSystem.cacheDirectory,
+};
+
+class BaseApiService {
```

Comment:
This is great, and a must-have for projects migrating to MEC going forward.
I am a little concerned about the tight coupling to App insights that is implemented in this class. In itself, a base API service should not include these things by default. I assume this forces every consumer of this code to also implement App Insights. Perhaps this is a business requirement, but not something we should enforce in this code.

I'd suggest decoupling the App Insights functionality from the api service and expose an API implementing wrappers ([Decorator pattern](https://refactoring.guru/design-patterns/decorator)) around the Base API Service methods instead.

Part of the reason I write this is since this base service would be great for implementing JS wrappers for Equinor API's in the future that all our apps can use (maintenance API is used by multiple apps in our portfolio for example). It would be awkward if the official Maintenance API JS package forced you to implement App Insights.

Reply:
I kinda agree, but not sure how to solve it in a good way. All of our iOS apps need to implement appInsights once this is merged in to main. It would be great to keep it optional, but the peer dependency packages are not optional. In our use case we kind of want app insights in all of our apps anyway. Maybe this could be a discussion for the frontend meeting?